common/src/main/java/org/apache/iceberg/common/DynMethods.java

```diff
@@ -51,10 +51,8 @@ public static class UnboundMethod {
           (method == null || method.isVarArgs()) ? -1 : method.getParameterTypes().length;
     }
 
-    /** @deprecated since 1.6.0, will be removed in 1.7.0 */
-    @Deprecated // will become private
     @SuppressWarnings("unchecked")
-    public <R> R invokeChecked(Object target, Object... args) throws Exception {
+    private <R> R invokeChecked(Object target, Object... args) throws Exception {
```

Comment:
> `// will become private`

my mistake
it was partially correct in a sense that it should stop being public API

but it seems i overlooked the fact the method is overridden (by `DynConstructors.Ctor`) in a way changing the behavior (invoking Method vs invoking a Constructor).
as such, it should become _package_-private, so that the override still works